src/main/scala/firrtl/options/StageAnnotations.scala

```diff
@@ -168,3 +168,17 @@ case object WriteDeletedAnnotation extends NoTargetAnnotation with StageOption w
   )
 
 }
+
+/** If this [[firrtl.annotations.Annotation Annotation]] exists in an [[firrtl.AnnotationSeq AnnotationSeq]], then
+  * Dump the raw [[firrtl.AnnotationSeq AnnotationSeq]] to a file on an error
+  */
+case object DumpOnCrashAnnotation extends NoTargetAnnotation with StageOption with HasShellOptions {
+
+  val options = Seq(
+    new ShellOption[Unit](
+      longOption =  "dump-on-crash",
+      toAnnotationSeq = { _ => Seq(this) },
+      helpText = "Dump raw Annotations to error.anno.json on error"
+    )
+  )
+}
```

Comment:
High level: this is an option that is associated with the `firrtl.options` package (which are intended to be common to all Stages, e.g., both `FirrtlStage` and `ChiselStage`). However, this is only handled inside a FIRRTL-specific phase wrapper, `CatchExceptions`. I think this means that this is then really a `FirrtlOption` specific to FIRRTL and not a `StageOption`.

Alternatively, this behavior could be something general to all stages and could be implemented as such.

Reply:
Thanks for clarifying the difference between `StageOption` and `FirrtlOption`.  It probably makes sense for this to be handled somewhere for `ChiselStage` as well.  I just happened to be. debugging something in firrtl this time.

Reply:
Chisel is a bit more monolithic and doesn't have a way of dumping partial circuits (at least currently), it would be nice though...

Reply:
So, for now should I move this back to being a  `FirrtlOption`?  I had it in there before I got distracted with `Viewer` and `StageOption` and decided to move it:
![shiny meme](https://i.chzbgr.com/full/428609792/hCE770C3D/ooh-shiny)

Reply:
Thoughts @seldridge?

Reply:
If it's *not* intended to be common to all stages, then I'd prefer to have it as a `FirrtlOption`. It should be a pretty straight migration, including the shiny typeclass stuff: https://github.com/freechipsproject/firrtl/blob/master/src/main/scala/firrtl/stage/package.scala#L23.